src/main/kotlin/apply/ui/admin/mail/MailFormView.kt

```diff
@@ -0,0 +1,180 @@
+import support.views.createSearchBar
+import support.views.createUploadButton
+
+@Route(value = "admin/emails", layout = BaseLayout::class)
+class MailFormView(
+    private val applicantService: ApplicantService,
+    private val recruitmentService: RecruitmentService,
+    private val evaluationService: EvaluationService,
+    private val mailTargetService: MailTargetService,
+    private val mailProperties: MailProperties
+) : BindingFormLayout<MailSendData>(MailSendData::class) {
+    private val subject: TextField = TextField("메일 제목")
+    private val content: TextArea = createMailBody()
+    private val mailTargetGrid: Grid<MailTargetResponse> = createMailTargetsGrid()
+    private val mailTargets: MutableSet<MailTargetResponse> = mutableSetOf()
+
+    init {
+        add(createTitle(), createMailForm())
+        setWidthFull()
+        setResponsiveSteps(ResponsiveStep("0", 1))
+        drawRequired()
+    }
+
+    private fun createTitle(): Component {
+        return HorizontalLayout(H1("메일 발송")).apply {
+            setSizeFull()
+            justifyContentMode = FlexComponent.JustifyContentMode.CENTER
+        }
+    }
+
+    private fun createMailForm(): Component {
+        val subjectText = VerticalLayout(H3("메일 제목"), subject).apply { setWidthFull() }
+        val sender = VerticalLayout(createSender())
+        val recipientFilter = VerticalLayout(createRecipientFilter())
+        val mailBody = VerticalLayout(content)
+        val uploadFile = createUploadButton("첨부파일", MultiFileMemoryBuffer()) {
+            /*
+            todo: 추후 업로드 된 파일을 메일로 첨부하는 로직이 추가되어야 함
+             (uploadFiles 같은 필드를 두고 mail을 보내는 기능에 포함시키면 될 것 같음)
+            it.files.forEach { fileName ->
+                val fileData = it.getFileData(fileName)
+                val inputStream = it.getInputStream(fileName)
+                val readBytes = inputStream.readBytes()
+            }
+            */
+        }
+        val sendButton = createMailSendButton()
+
+        return VerticalLayout(
+            subjectText,
+            sender,
+            recipientFilter,
+            VerticalLayout(mailTargetGrid),
+            mailBody,
+            VerticalLayout(uploadFile),
+            sendButton
+        ).apply {
+            setSizeFull()
+            justifyContentMode = FlexComponent.JustifyContentMode.CENTER
+            defaultHorizontalComponentAlignment = FlexComponent.Alignment.CENTER
+        }
+    }
+
+    private fun createMailTargetsGrid(): Grid<MailTargetResponse> {
+        return Grid<MailTargetResponse>(10).apply {
+            addSortableColumn("이름", MailTargetResponse::name)
+            addSortableColumn("이메일", MailTargetResponse::email)
+            addColumn(createRemoveButtonRender())
+        }
+    }
+
+    private fun createRemoveButtonRender(): Renderer<MailTargetResponse> {
+        return ComponentRenderer<Component, MailTargetResponse> { response ->
+            createPrimarySmallButton("삭제") {
```

Comment:
```suggestion
            createErrorButton("삭제") {
```
여기는 ErrorButton을 쓰면 좋을 것 같네요 ㅎㅎ

Reply:
처음엔 엄청 신경 썻던 부분인데... 어느 순간 별 생각없이 버튼을 만들고 있었네요 🙄